client/dashboard/data/site-atomic-transfers.ts

```
@@ -0,0 +1,29 @@
+import wpcom from 'calypso/lib/wp';
+
+export type AtomicTransferStatus =
+	| 'pending'
+	| 'active'
+	| 'provisioned'
+	| 'completed'
+	| 'error'
+	| 'reverted'
+	| 'relocating_revert'
+	| 'relocating_switcheroo'
+	| 'reverting'
+	| 'renaming'
+	| 'exporting'
+	| 'importing'
+	| 'cleanup';
+
+export interface AtomicTransfer {
+	status: AtomicTransferStatus;
+}
+
+export async function fetchLatestAtomicTransfer(
+	siteId: number
+): Promise< AtomicTransfer | undefined > {
```

This is an interesting return result. Have you observed the endpoint returning an empty body?

---

The endpoint will return 404 if there is no any atomic transfers, in which case this function will return undefined (and so will the query's `data`).

---

Sorry I'm wrong. In this case the function will throw instead of returning 🤦 Removed the `undefined`!